packages/bot-web-ui/src/stores/load-modal-store.ts

```diff
@@ -200,77 +267,88 @@ export default class LoadModalStore {
 
         this.setActiveTabIndex(0); // Reset to first tab.
         this.setLoadedLocalFile(null);
-    }
+    };
 
     @action.bound
-    onZoomInOutClick(is_zoom_in) {
+    onZoomInOutClick = (is_zoom_in: string): void => {
         if (this.preview_workspace) {
             this.preview_workspace.zoomCenter(is_zoom_in ? 1 : -1);
         }
-    }
+    };
 
     @action.bound
-    previewRecentStrategy(workspace_id) {
+    previewRecentStrategy = (workspace_id: string): void => {
         this.setSelectedStrategyId(workspace_id);
-
         if (!this.selected_strategy) {
             return;
         }
 
-        if (!this.recent_workspace || !this.recent_workspace.rendered) {
-            const ref = document.getElementById('load-strategy__blockly-container');
+        const ref = document.getElementById('load-strategy__blockly-container');
 
-            if (!ref) {
-                // eslint-disable-next-line no-console
-                console.warn('Could not find preview workspace element.');
-                return;
-            }
-
-            this.recent_workspace = Blockly.inject(ref, {
-                media: `${__webpack_public_path__}media/`,
-                zoom: {
-                    wheel: true,
-                    startScale: config.workspaces.previewWorkspaceStartScale,
-                },
-                readOnly: true,
-                scrollbars: true,
-            });
+        if (!ref) {
+            // eslint-disable-next-line no-console
+            console.warn('Could not find preview workspace element.');
+            return;
         }
-
-        load({ block_string: this.selected_strategy.xml, drop_event: {}, workspace: this.recent_workspace });
-    }
+        clearInjectionDiv();
+        this.recent_workspace = Blockly.inject(ref, {
+            media: `${__webpack_public_path__}media/`,
+            zoom: {
+                wheel: true,
+                startScale: config.workspaces.previewWorkspaceStartScale,
+            },
+            readOnly: true,
+            scrollbars: true,
+        });
+        if (this.is_load_modal_open === true) {
+            //load({ block_string: this.selected_strategy.xml, drop_event: {}, workspace: this.recent_workspace });
+            //return false;
```

Comment:
kept this commented because there is an issue when we load the popup the load method fires continuously here. tried to use return false but it doen't seem to work

Reply:
toggleLoadModal is responsible to change the value of `is_load_modal_open ` but I could not find where is exactly being used. Maybe this is ten root of problem